sklearn/model_selection/tests/test_search.py

```diff
@@ -1846,3 +1839,61 @@ def test_search_cv__pairwise_property_equivalence_of_precomputed():
 
     attr_message = "GridSearchCV not identical with precomputed metric"
     assert (preds_original == preds_precomputed).all(), attr_message
+
+
+@pytest.mark.parametrize(
+    "SearchCV, param_search",
+    [(GridSearchCV, {'a': [0.1, 0.01]}),
+     (RandomizedSearchCV, {'a': np.random.randint(1, 3, size=2)})]
+)
+def test_scalar_fit_param(SearchCV, param_search):
+    # check general support for scalar in fit_params
+    # non-regression test for:
+    # https://github.com/scikit-learn/scikit-learn/issues/15805
+    class TestEstimator(BaseEstimator, ClassifierMixin):
+        def __init__(self, a=None):
+            self.a = a
+
+        def fit(self, X, y, r=None):
+            assert r == 42
+
+        def predict(self, X):
+            return np.zeros(shape=(len(X)))
+
+    model = SearchCV(TestEstimator(), param_search)
+    X, y = make_classification(random_state=42)
+    model.fit(X, y, r=42)
+
+
+def _custom_lgbm_metric(y_test, y_pred):
+    # y_pred are probablities which need to be thresholded
+    y_pred = (y_pred > 0.5).astype(int)
+    acc = accuracy_score(y_test, y_pred)
+    # required output of format: (eval_name, eval_result, is_higher_better)
+    return ('accuracy', acc, True)
+
+
+@pytest.mark.parametrize("metric", ['auc', _custom_lgbm_metric])
+@pytest.mark.parametrize(
+    "SearchCV, param_search",
+    [(GridSearchCV, {'learning_rate': [0.1, 0.01]}),
+     (RandomizedSearchCV, {'learning_rate': uniform(0.01, 0.1)})]
+)
+def test_scalar_fit_param_lgbm(metric, SearchCV, param_search):
```

Comment:
This code is not covered because we don't have a build with `lightgbm`. I thought we did to test part of the code in `HistGradientBoostingClassifier`. Would it make sense to add one to test these tests?

Reply:
This would be nice to have. I do not recall the original reason for not including `lightgbm` in the test suite.

Reply:
If we had such a test, we should `pip install lightgbm` on at least one CI worker.

Reply:
The benefit: we will detect such downstream breakage earlier in the future.

The cons: if there is any upstream packaging issue / API change in lightgbm our CI will start to fail and that add-up to our maintenance cost.